### PR TITLE
Add NXP's i.MX8QX SoC support

### PR DESCRIPTION
**Pull request by Anson-Huang**
This patch set adds NXP's i.MX8QX SoC support, i,MX8QX is a ARMv8 SoC with 4 Cortex-A35 cores and 1 system controller (Cortex-M4) inside, with this patch set, i.MX8QX SMP Linux kernel can boot up successfully, using reset to BL31 method to boot up u-boot and then Linux kernel.

### CODE REVIEW

**Comment by dp-arm** [comment]
IP review: +1

File: plat/imx/imx8qx/imx8qx_bl31_setup.c [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+#include <sci/sci.h>
+#include <sec_rsrc.h>
+#include <stdbool.h>
+#include <xlat_tables.h>
+
+#define BL31_COHERENT_RAM_BASE (unsigned long)(&__COHERENT_RAM_START__)
+#define BL31_COHERENT_RAM_LIMIT (unsigned long)(&__COHERENT_RAM_END__)
+#define BL31_RO_BASE	(unsigned long)(&__RO_START__)
+#define BL31_RO_LIMIT	(unsigned long)(&__RO_END__)
+#define BL31_END	(unsigned long)(&__BL31_END__)
```
Comment (antonio-nino-diaz-arm) [inline plat/imx/imx8qx/imx8qx_bl31_setup.c]: Use:
```
IMPORT_SYM(unsigned long, __COHERENT_RAM_START__, BL31_COHERENT_RAM_BASE);
```
and so on.

File: plat/imx/imx8qx/imx8qx_bl31_setup.c [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+}
+
+void bl31_plat_arch_setup(void)
+{
+	mmap_add_region(BL31_BASE, BL31_BASE, BL31_END - BL31_BASE, MT_MEMORY | MT_RW);
+	mmap_add_region(BL31_BASE, BL31_BASE, BL31_RO_LIMIT - BL31_RO_BASE, MT_MEMORY | MT_RO);
+	mmap_add_region(IMX_BOOT_UART_BASE, IMX_BOOT_UART_BASE, IMX_BOOT_UART_SIZE, MT_DEVICE | MT_RW);
+	mmap_add_region(SC_IPC_BASE, SC_IPC_BASE, SC_IPC_SIZE, MT_DEVICE | MT_RW);
+	mmap_add_region(PLAT_GICD_BASE, PLAT_GICD_BASE, PLAT_GICD_SIZE, MT_DEVICE | MT_RW);
+	mmap_add_region(PLAT_GICR_BASE, PLAT_GICR_BASE, PLAT_GICR_SIZE, MT_DEVICE | MT_RW);
```
Comment (antonio-nino-diaz-arm) [inline plat/imx/imx8qx/imx8qx_bl31_setup.c]: You could have the regions as an array of `mmap_region_t` structs and use `mmap_add()` instead.

File: plat/imx/imx8qx/imx8qx_bl31_setup.c (hunk 1 above)
Comment (Anson-Huang) [reply plat/imx/imx8qx/imx8qx_bl31_setup.c]: Thanks for view, will improve it in V2 pull request.

File: plat/imx/imx8qx/imx8qx_bl31_setup.c (hunk 2 above)
Comment (Anson-Huang) [reply plat/imx/imx8qx/imx8qx_bl31_setup.c]: Thanks for view, will improve it in V2 pull request, and I will fix the copyright year and checkpatch error together.

**Comment by Anson-Huang** [comment]
I fixed all the review comments, also add i.MX8QM support, will created a new pull request to cover NXP's i.MX8QXP and i.MX8QM SoC support, thanks all for reviewing this patch set.

**Pull request closed without merging**